src/components/SelectChildForClass.tsx

```diff
@@ -12,6 +13,12 @@ type SelectChildForClassProps = {
 export default function SelectChildForClass(
     props: SelectChildForClassProps,
 ): JSX.Element {
+    if (!props.eligible[props.selectedChild]) {
+        let index = 0;
+        while (!props.eligible[index] && index < props.eligible.length) index++;
+        props.setSelectedChild(index);
+    }
+
```

Comment:
so i needed to add a check so that the default `selectedChild` on page load was actually an eligible child -- this works UNLESS no children are eligible for a particular class, in which then the code breaks (assumingly because the index goes out of bounds) 

the way the current flow is set up (as the `selectedChild` is passed onto the next page), i'm not sure if there's a way around it... another option could to be if all children are ineligible, we direct to a different page saying that none of their children are eligible for this program?

Reply:
doing a `router.back()` + error toast message makes sense to me!

Reply:
i just realized that if no children are eligible, then the program won't be clickable in the first place  (see below), so do we still need a router call? The only time where there'd be an issue is if the user tried to directly go to the link (http://localhost:3000/parent/enrollment?classId=10001) instead of clicking to it -- 

<img width="1113" alt="Screen Shot 2021-11-26 at 12 24 30 AM" src="https://user-images.githubusercontent.com/45214377/143531179-0695be96-c353-45f9-bf5e-30dea2c990d2.png">

I tried doing a router thing anyways to see if it'd work but it's giving me an error -- when we call a router.back() does it break away from the rest of the code? Because even with a router.back() call to home, I still get an error in the `ClassEnrollmentConfirm` page.

Reply:
Yes, we always want to consider all edge cases like you described. Using router takes some time to load (async), so what I usually do is in the clause where we are routing to somewhere, I also return a <CommonLoading /> like:
```typescript
router.back();
return <CommonLoading />
```

to avoid reaching the "bad" that we are trying to avoid by using router.back().

Reply:
there's no point of a toast then, right? because it won't always show depending on what it's routing back to (eg. i went to the URL directly from a google page so no toast showed up)

Reply:
mm, yea that's true. Let's leave the toast out!